src/content/docs/en/guides/deploy/cloudflare.mdx

```diff
@@ -1,121 +1,163 @@
 
-### Quick install
+You can also use a CI/CD system such as [Workers Builds (BETA)](https://developers.cloudflare.com/workers/ci-cd/builds/) to automatically build and deploy your site on push. 
 
-Add the Cloudflare adapter to enable SSR in your Astro project with the following `astro add` command. This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step.
+If you're using Workers Builds:
 
-```bash
-npx astro add cloudflare
-```
+<Steps>
+1. Follow Steps 1-3 from the Wrangler section above.
+2. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/) and navigate to `Workers & Pages`. Select `Create`.
+3. Under `Import a repository`, select a Git account and then the repository containing your Astro project.
+4. Configure your project with:
+    - Build command: `npx astro build`
+    - Deploy command: `npx wrangler deploy`
+5. Click `Save and Deploy`. You can now preview your Worker at its provided `workers.dev` subdomain.
+</Steps>
 
-### Manual install
+## Cloudflare Pages
 
-If you prefer to install the adapter manually instead, complete the following two steps:
+### How to deploy with Wrangler
 
 <Steps>
-1. Add the `@astrojs/cloudflare` adapter to your project's dependencies using your preferred package manager. If you're using npm or aren't sure, run this in the terminal:
-
+1. Install [Wrangler CLI].(https://developers.cloudflare.com/workers/wrangler/get-started/).
     ```bash
-    npm install @astrojs/cloudflare
+      npm install wrangler@latest --save-dev
     ```
 
-2. Add the following to your `astro.config.mjs` file:
+2. If your site uses on demand rendering, install the `@astrojs/cloudflare` adapter. 
+    This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step.
+      ```bash
+        npx astro add cloudflare
+      ```
+    <ReadMore>Read more about [on demand rendering in Astro](/en/guides/on-demand-rendering/).</ReadMore>
 
-    ```js title="astro.config.mjs" ins={2, 5-6}
-    import { defineConfig } from 'astro/config';
-    import cloudflare from '@astrojs/cloudflare';
+3. Preview your project locally with Wrangler.
+    ```bash
+      npx astro build && npx wrangler pages dev ./dist
+    ```
 
-    export default defineConfig({
-      output: 'server',
-      adapter: cloudflare()
-    });
+4. Deploy using `npx wrangler deploy`.
+    ```bash
+      npx astro build && npx wrangler pages deploy ./dist
     ```
 </Steps>
 
-<ReadMore>Read more about [SSR in Astro](/en/guides/on-demand-rendering/).</ReadMore>
+After your assets are uploaded, Wrangler will give you a preview URL to inspect your site.
+
+
+### How to deploy a site with Git
+
+<Steps>
+1. Push your code to your git repository (e.g. GitHub, GitLab).
+
+2. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/) and navigate to `Workers & Pages`. Select `Create` and then select the `Pages` tab. Connect your git repository.
```

Comment:
I thought that we want a generic instruction instead of a detailed one? As mentioned here: https://github.com/withastro/docs/pull/10790#issuecomment-2630828967

Reply:
@vrabe Ooh, you're right, we missed this one! Would you like to make a PR to update with your suggestion?